web/src/lightbox.ts

```diff
@@ -261,7 +263,20 @@ function display_image(payload: Payload): void {
     $(".image-preview, .media-actions, .media-description, .download, .lightbox-zoom-reset").show();
 
     const $img_container = $("#lightbox_overlay .image-preview > .zoom-element");
-    const $img = $("<img>").attr("src", payload.source);
+    const $img = $("<img>");
+    if (payload.preview && payload.original_width_px && payload.original_height_px) {
+        $img.attr("src", payload.source);
+        $img.css({
+            width: `calc(min(${payload.original_width_px}px, 100%`,
```

Comment:
This is a workable approach, and in testing with images of varying aspect ratios, it seems to deliver as expected.

One alternative to this I could imagine would be to set explicit `height` and `width` attributes on the `<img>` tag itself, which is [generally good practice again](https://www.smashingmagazine.com/2020/03/setting-height-width-images-important-again/), and then use `payload.original_width_px` `payload.original_height_px` to set the aspect ratio.

What you'd then have is something like:

```js
$img.attr("width", payload.original_width_px);
$img.attr("height", payload.original_height_px);

$img.css({
  // As before; might not even need this in JS,
  // as there are no variables here.
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
});
```

I've prepared [a Codepen showing this technique](https://codepen.io/karlstolley/pen/oNrLmPw). The image there is originally 2848 by 4288, but I've mocked the preview image by multplying those values by 100, just to pretend the already-large image there is a placeholder for something even larger.